### ISSUE TITLE
Make the README a kinder place by removing/replacing the John Woods quote

### ISSUE THREAD

**Issue by li-boxuan** [coala/coala#5792]
This is migrated from https://github.com/coala/coala/pull/5767

As @joenash suggests,

```
Hey folks 👋

This PR is to discuss the removal of the John Woods quote from the README. I believe that quote is a product of its time (1991), and that Coala can do better to be a welcoming project.
We should write maintainable code because we have respect for our peers and colleagues who will read and work with it, not because of threat of death. The quote is also unnecessarily gendered.

Apologies also for the drive-by PR 🙇 For some additional context, I became aware of this quote because I work with a lot of software development students, and saw a large number were contributors to Coala. Coala is doing an amazing job of being accessible to students and upcoming software developers, but this also makes it important to ensure that students who work on this project are being taught respect for their fellow humans 💖

I've added [WIP] as currently this commit removes the quote entirely. It may be more suitable to find an alternative to fill that space and act as the one-line pitch. If anyone has suggestions, please feel free to add them!
```

Some follow-up discussions can be found on that PR. This issue is opened as the pull request is not a correct place to discuss/vote.

**Comment by jayvdb** [comment]
This quote came out of a README jam session where we removed lots of complexity from the README and looked for something which communicated "what is coala" a bit better.

IIRC, I proposed it, or endorsed it, and was involved in tracking down the source so it was not '-- anonymous'.

I dont recall there being a lot of other quote suggestions, so it is quite possible that there are better ones.

The quote was chosen as it typifies *why* we build coala, not necessarily *how* we build coala.

We build it because we need these tools to catch our worse instincts to write sloppy code.  We all do it.  All the time.  Maybe not when writing in our preferred languages, but when we have to touch some file we dont care much about.  However, someone else cares about that file format, and they would like the coder and reviewer and merger to all 'hold the line' even if the primary owner is away.

So we build tools instead of writing coding style guidelines and having pointless ongoing discussions about style. ;-).  And we implement new checks as regularly as can to continue to lift the bar higher, while also giving newcomers meaningful error messages that they can use to understand how to contribute a mergable MR.

Respect for peers is an interesting point. Sloppy code is often a byproduct of a surface-thin respectful environment, as reviewers can tend to avoid confrontation about little details because it is much harder to demand high quality and do coaching and newbie induction, etc, etc.  It can be very hard to keep pressing for small little fixes when a newbie has come a long way, and the PR is over 100 comments, and everyone involved wants it to be merged.  Merging lots of little degradation of style/quality means maintainers need to be constantly doing fixups themselves, causing burnout, or gradually the codebase rots on the inside while looking great on the outside.  tools+bots dictating accepted style, minimum quality, etc remove much of the confrontation out of PRs.  That is perhaps a part of the 'how' we build coala.  This is similar to the golang goal of "gofmt is the standard", but inverse approach.

Could we provide more context to the existing quote, separating why / how?  Yes.

Can we get a better quote (or create our own)?  Yes.  Suggestions?

Removing it without a replacement is also a viable option.

**Comment by joenash** [comment]
Thanks for that context @jayvdb, that's awesome!

@palash25 made some great suggestions in the PR thread for some alternatives, though I think we can do better than Linus (again, not the most welcoming figure in OSS, despite his huge technical achievements) and Martin Fowler (he's chosen some interesting hills to die on as of late). 

But again - this is just my thoughts as someone whose profession is helping to build developer communities, and not the input of someone who is part of the coala project. I recognise that y'all who are part of the project may disagree with my precise opinions on the above individuals, and I wouldn't for a minute wish to imply that my opinion here carries any weight vs the regular contributors to this project 💖

**Comment by jayvdb** [comment]
I definitely oppose "given enough eyeballs, all bugs are shallow" and derivatives.

While it is seems true for some projects, there are very few projects where it is applicable.

It doesnt help the typical OSS project which coala aims to help with, as they usually have few eyeballs with experience, and the eyeballs are strained with other responsibilities.

The Martin Fowler quote:

:"I’m not a great programmer; I’m just a good programmer with great habits."

.. is more relevant, and it touches a bit on the how and why, and is a bit positive rather than threatening death by psychopath, which is probably a good thing. However when used as a quote, properly attributed, it is aloft.  It is about someone great, self-deprecating, but we all read the self deprecation as modesty.  Not about us ordinary coders.  It gives no clue as to what good habits propelled him to be great.

His quote:

:"Any fool can write code that a computer can understand. Good programmers write code that humans can understand."

... is more apt IMO, about the pursuit of writing code that communicates clearly, reducing strain on reviewers, maintainers and archeologists.

(And using that quote provides a little irony in that coala's pursuit is trying to get the computer to not be fooled by the code that people write in practise, and often it is well-written code which provides the biggest hurdles for static analysis and beautifiers.)

But more suggestions are needed.

**Comment by snow8gaia** [comment]
I don't find that quote making the readme … cruel. 😂

**Comment by snow8gaia** [comment]
And as for this:-
>We should write maintainable code because we have respect for our peers and colleagues who will read and work with it, not because of threat of death.

I think this sounds like a bad replacement. Please don't use this. It will confuse newcomers. And repel some who avoid everything political. 

And the quote does not threaten anyone with death. Not unless its intention is misconstrued.

**Comment by snow8gaia** [comment]
>"Any fool can write code that a computer can understand. Good programmers write code that humans can understand."

**Comment by li-boxuan** [comment]
> Is the discussion happening on a separate channel?

No, the discussion is happening here. Feel free to leave more thoughts.

**Comment by palash25** [comment]
> **We should write maintainable code because we have respect for our peers and colleagues who will read and work with it, not because of threat of death.**

> I think this sounds like a bad replacement. Please don't use this. It will confuse newcomers

This wasn't suggested as a replacement for the current quote, Joe was just making his point, so no one is considering that as an option. @snow8gaia 
Please read the original thread https://github.com/coala/coala/pull/5767 to get more context on this

**Comment by joenash** [comment]
> If anything this quote is the last thing I care about, I just want to setup this repo on my Windows 10 machine

Considering it’s ”the last thing you care about”, you have a lot of strong opinions about keeping the quote 😉

**Comment by snow8gaia** [comment]
>No, the discussion is happening here. Feel free to leave more thoughts.

Thanks a lot.

**Comment by snow8gaia** [comment]
>no one is considering that as an option

Thanks for the info.

**Comment by snow8gaia** [comment]
Correcting @snow8gaia 
>It will confuse newcomers.

It will make newcomers think this is a place to avoid. Wait! What am I doing? LOL.

@palash25 Did you read https://github.com/coala/coala/issues/5792#issuecomment-440338592  it will clear up the confusion.

**Comment by snow8gaia** [comment]
>Please read the original thread #5767 to get more context on this

At the time, I did not know this thread was connected to another thread. Or any thread. 

I read #5767 . Thanks.

**Comment by snow8gaia** [comment]
>Considering it’s ”the last thing you care about”, you have a lot of strong opinions about keeping the quote 

LOL. 


Yeah, @joenash I talk a lot, so it might seem strong. I usually avoid this like most people.
 I guess I wanted to get a clear understanding of how this community works.

**Comment by snow8gaia** [comment]
So, anyway what is the  status on this?

**Comment by palash25** [comment]
> I guess I wanted to get a clear understanding of how this community works.

You can get a much better understanding of that by contributing to coala and following our [Newcomer](http://api.coala.io/en/latest/Developers/Newcomers_Guide.html) process :wink:
